Build error with HPX_WITH_DYNAMIC_HPX_MAIN=OFF

## Actual Behavior
I get the following error message when building HPX with these cmake flags: `-DHPX_WITH_THREAD_IDLE_RATES=ON -DHPX_WITH_THREAD_LOCAL_STORAGE=ON -DHPX_WITH_DYNAMIC_HPX_MAIN=OFF`, which is resolved when I remove `-DHPX_WITH_DYNAMIC_HPX_MAIN=OFF`.
```
[ 68%] Built target throttle_component
Scanning dependencies of target 1d_stencil_replay_validate
../../../lib/libhpx_wrap.a(main.cpp.o): In function `hpx::init(int, char**, hpx::init_params const&)':
main.cpp:(.text._ZN3hpx4initEiPPcRKNS_11init_paramsE[_ZN3hpx4initEiPPcRKNS_11init_paramsE]+0x4c): undefined reference to `hpx_main(boost::program_options::variables_map&)'
Scanning dependencies of target async_replicate_vote
../../../lib/libhpx_wrap.a(main.cpp.o): In function `hpx::init(int, char**, hpx::init_params const&)':
main.cpp:(.text._ZN3hpx4initEiPPcRKNS_11init_paramsE[_ZN3hpx4initEiPPcRKNS_11init_paramsE]+0x4c)clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation): undefined reference to `
hpx_main(boost::program_options::variables_map&)'
make[2]: *** [bin/response_file] Error 1
make[1]: *** [libs/program_options/examples/CMakeFiles/response_file.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
../../../lib/libhpx_wrap.a(main.cpp.o): In function `hpx::init(int, char**, hpx::init_params const&)':
main.cpp:(.text._ZN3hpx4initEiPPcRKNS_11init_paramsE[_ZN3hpx4initEiPPcRKNS_11init_paramsE]+0x4c): undefined reference to `hpx_main(boost::program_options::variables_map&)'
[ 68%] Built target component_storage_component
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/custom_syntax] Error 1
../../../lib/libhpx_wrap.a(main.cpp.o): In function `hpx::init(int, char**, hpx::init_params const&)':
main.cpp:(.text._ZN3hpx4initEiPPcRKNS_11init_paramsE[_ZN3hpx4initEiPPcRKNS_11init_paramsE]+0x4c): undefined reference to `hpx_main(boost::program_options::variables_map&)'
../../../lib/libhpx_wrap.a(main.cpp.o): In function `hpx::init(int, char**, hpx::init_params const&)':
main.cpp:(.text._ZN3hpx4initEiPPcRKNS_11init_paramsE[_ZN3hpx4initEiPPcRKNS_11init_paramsE]+0x4c): undefined reference to `hpx_main(boost::program_options::variables_map&)'
make[1]: *** [libs/program_options/examples/CMakeFiles/custom_syntax.dir/all] Error 2
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/config_file_types] Error 1
make[1]: *** [libs/program_options/examples/CMakeFiles/config_file_types.dir/all] Error 2
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/option_groups] Error 1
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
../../../lib/libhpx_wrap.a(main.cpp.o): In function `hpx::init(int, char**, hpx::init_params const&)':
main.cpp:(.text._ZN3hpx4initEiPPcRKNS_11init_paramsE[_ZN3hpx4initEiPPcRKNS_11init_paramsE]+0x4c): undefined reference to `hpx_main(boost::program_options::variables_map&)'
make[2]: *** [bin/options_hierarchy] Error 1
make[1]: *** [libs/program_options/examples/CMakeFiles/option_groups.dir/all] Error 2
make[1]: *** [libs/program_options/examples/CMakeFiles/options_hierarchy.dir/all] Error 2
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/regex] Error 1
make[1]: *** [libs/program_options/examples/CMakeFiles/regex.dir/all] Error 2

```


## Specifications

  - HPX Version: current master, commit id `7a07598fbfcb4a871d16d568e31dbd211dc74d4f`
  - Platform (compiler, OS): Linux, clang 6.0.1, boost 1.68

--- a/wrap/CMakeLists.txt
+++ b/wrap/CMakeLists.txt
@@ -79,6 +79,7 @@ add_library(
 )
 set_target_properties(hpx_wrap PROPERTIES POSITION_INDEPENDENT_CODE ON)
 target_link_libraries(hpx_wrap PRIVATE hpx_core)
+target_link_libraries(hpx_wrap PRIVATE hpx_init)
 target_link_libraries(hpx_wrap PRIVATE hpx_private_flags)
 target_compile_definitions(hpx_wrap PRIVATE HPX_APPLICATION_EXPORTS)
 target_include_directories(

